x/application/keeper/msg_server_undelegate_from_gateway_test.go

```diff
@@ -92,6 +92,9 @@ func TestMsgServer_UndelegateFromGateway_SuccessfullyUndelegate(t *testing.T) {
 	_, err = srv.UndelegateFromGateway(ctx, undelegateMsg)
```

Comment:
I know you said we don't need any new unit/integration tests, but I still feel like a unique test where we do stuff:

1. Undelegate mid session
2. Redelegate mid session
3. Check when we're halfway into the retention period (still there)
4. Check when we're past the retention period (shouldn't be there)

I STRONGLY feel there are at least 3-4 "unit" tests that need to be added for very cryptic logic like this.

If you disagree or need help with framing them, lmk